launcher/src/main/kotlin/com/cognifide/gradle/aem/launcher/BuildScaffolder.kt

```diff
@@ -1,34 +1,28 @@
 package com.cognifide.gradle.aem.launcher
 
+import java.io.File
 import java.util.*
 
 class BuildScaffolder(private val launcher: Launcher) {
 
     fun scaffold() {
-        saveBuildSrc()
         saveProperties()
         saveSettings()
         saveRootBuildScript()
-        saveEnvBuildScript()
-    }
 
-    private fun saveBuildSrc() = launcher.workFileOnce("buildSrc/build.gradle.kts") {
-        println("Saving Gradle build source script file '$this'")
-        writeText("""
-            repositories {
-                mavenLocal()
-                mavenCentral()
-                gradlePluginPortal()
-            }
-            
-            dependencies {
-                implementation("com.cognifide.gradle:aem-plugin:${launcher.pluginVersion}")
-                implementation("com.cognifide.gradle:common-plugin:1.0.41")
-                implementation("com.neva.gradle:fork-plugin:7.0.5")
-            }
-        """.trimIndent())
+        if (propertyAemVersion()?.isNotBlank() == true) {
+            if (propertyAemVersion() == "cloud") EnvCloudScaffolder(launcher).scaffold() else EnvOnPremScaffolder(launcher).scaffold()
+        } else {
+            EnvScaffolder(launcher).scaffold()
+        }
     }
 
+    fun propertyAemVersion() = archetypeProperties()?.getProperty("aemVersion")
+
+    fun archetypeProperties() = if (File("archetype.properties").exists()) Properties().apply {
```

Comment:
introduce a `val archetypeProperties get() = File("archetype.properties")` then reuse it